hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java

```diff
@@ -281,13 +282,20 @@ protected NioNetworking initNetworking() {
     }
 
     private WaitStrategy initializeWaitStrategy(ClientConfig clientConfig) {
-        ClientConnectionStrategyConfig connectionStrategyConfig = clientConfig.getConnectionStrategyConfig();
-        ConnectionRetryConfig expoRetryConfig = connectionStrategyConfig.getConnectionRetryConfig();
-        return new WaitStrategy(expoRetryConfig.getInitialBackoffMillis(),
-                expoRetryConfig.getMaxBackoffMillis(),
-                expoRetryConfig.getMultiplier(),
-                expoRetryConfig.getClusterConnectTimeoutMillis(),
-                expoRetryConfig.getJitter(), logger);
+        ConnectionRetryConfig retryConfig = clientConfig
+                .getConnectionStrategyConfig()
+                .getConnectionRetryConfig();
+
+        long clusterConnectTimeout = retryConfig.getClusterConnectTimeoutMillis();
+        if (failoverConfigProvided && !ConnectionRetryConfigAccessor.isClusterConnectTimeoutConfigured(retryConfig)) {
```

Comment:
I used that accessor approach to respect the configuration when the user sets an infinite timeout in a scenario like below.
```java
        ClientFailoverConfig failoverConfig = new ClientFailoverConfig();
        ClientConfig config = new ClientConfig();
        config.getConnectionStrategyConfig().getConnectionRetryConfig().setClusterConnectTimeoutMillis(Long.MAX_VALUE);
        failoverConfig.addClientConfig(config);
        HazelcastClient.newHazelcastFailoverClient(failoverConfig);
```
If we think that this scenario is unlikely or does not make sense, I can ditch that accessor and check whether or not cluster connect timeout equals to the default value for it.

Reply:
Yes, it looks like a wrong configuration. We may fail fast in this case. But what happens if user sets Int.MAX for example. This probably a wrong setting also.So I am not sure about failfast behaviour. Lets leave it aside.  

How you considered making the default value -1 ? We use that approach in a couple of places.
Documentation will make it clear that, in normal setup, we default to not closing the client. 
In failover we use 120 seconds. 

That way you can track if user set it or not without accessor.

Reply:
Okay, changed the default value to -1 and added documentation to the getter and setter methods. Can you take another look?